internal/runner/accept.go

```diff
@@ -121,27 +127,19 @@ func (r *Runner) accept(ctx context.Context, id string) error {
 	// Wait for an assignment
 	log.Info("waiting for job assignment")
 
-	var (
-		acceptTimer *time.Timer
-		timerMu     sync.Mutex
-		accepted    bool
-		canceled    bool
-	)
-
-	if r.acceptTimeout != 0 {
+	// If we have a timeout, then we setup a timer for accepting.
+	var acceptTimer *time.Timer
+	var canceled int32
+	if r.acceptTimeout > 0 {
 		acceptTimer = time.AfterFunc(r.acceptTimeout, func() {
-			timerMu.Lock()
-			defer timerMu.Unlock()
+			log.Error("runner timed out waiting for a job",
+				"timeout", r.acceptTimeout.String())
 
-			// If we raced to this point and timed out BUT the job was accepted, don't cancel
-			// the context.
-			if accepted {
-				return
-			}
+			// Mark that we canceled
+			atomic.StoreInt32(&canceled, 1)
 
-			canceled = true
```

Comment:
This was a data race in the prev code. `canceled` read/write was not protected by a mutex below. Not a huge issue and never caused any issue in practice, but worth noting it isn't strictly correct.